extension/ext-comments.js

```diff
@@ -755,7 +755,8 @@ function replaceComments(rootElem, comments, options=REPLACE_COMMENTS_DEFAULT_OP
       addCommentLink.href = '#';
     }
 
-    const orderDiv = createElement(holderDiv, 'div', 'comment-order', 'Order: ');
+    const orderDiv = createElement(holderDiv, 'div', undefined, 'Order: ');
+    orderDiv.id = 'comment-order';
```

Comment:
I would strongly prefer not to introduce global ids for dynamically created elements.

Instead, make `orderDiv` a global variable like you did with `commentListRoot`.

(Global variables aren't great either, but the current solution with assigning an id just so you can retrieve the element from the DOM later is basically a global variable in disguise. In that case I'd prefer the global variable to be explicit.)

Reply:
I can change this, but is there a reason to avoid ids? It seems to me that semantically that's the proper way to represent something there's exactly one of, like the order toggle or even the comment container.

Reply:
It's mostly personal preference, but there are also technical arguments in favor of it:

  - If the element is assigned to a global variable, then I can easily find references to it in the IDE. I think it's generally useful to keep code discoverable this way, which is why I prefer static approaches over dynamic ones.

  - Global ids can clash, just like global variable names. Remember that the extension doesn't own the DOM tree; it's created by Substack, so in theory, Substack can introduce an element with a conflicting id, which would break the extension. It's easy to reduce that risk to practically 0 by adding a specific id prefix (e.g. `acxi-`) but I'd rather avoid the issue entirely out of principle.

Overall, I'm trying to keep the code modular. For example, there is no fundamental reason why there couldn't be two ExtCommentListComponents on the same page, and right before your introduction of the `commentListRoot` global variable, that indeed worked. To keep components modular this way they should avoid global ids.

(And this is one of the things I might want to refactor in the future: I don't think `commentListRoot` *needs* to be global, we just need a better way to connect the options to the comment list instance. But it's not high priority to fix this.)